core/trino-main/src/main/java/io/trino/cost/TableScanStatsRule.java

```diff
@@ -55,13 +51,13 @@ public Pattern<TableScanNode> getPattern()
     }
 
     @Override
-    protected Optional<PlanNodeStatsEstimate> doCalculate(TableScanNode node, StatsProvider sourceStats, Lookup lookup, Session session, TypeProvider types)
+    protected Optional<PlanNodeStatsEstimate> doCalculate(TableScanNode node, StatsProvider sourceStats, Lookup lookup, Session session, TypeProvider types, TableStatsProvider tableStatsProvider)
```

Comment:
Why `StatsProvider` cannot handle caching? It does already, right? Something doesn't add up here.

Reply:
that was my initial idea as well. see https://github.com/trinodb/trino/pull/12196#discussion_r874660189 .